### PR TITLE
feat: Implement MultiLocation support in the deposit method in EVM and Substrate

### PR DESCRIPTION
**Pull request by enemycnt**
## Related Issue Or Context

Closes: #205 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [ ] I have added tests to cover my changes.
- [ ] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: packages/sdk/src/utils/helpers.ts [hunk 1]
```diff
@@ -94,18 +107,32 @@ export const constructMainDepositData = (
...
+  const recipientAddressInBytes = getRecipientAddressInBytes(recipientAddress);
   const depositDataBytes = constructMainDepositData(convertedAmount, recipientAddressInBytes);
   const depositData = utils.hexlify(depositDataBytes);
+
   return depositData;
 };
 
+/**
+ * Converts a recipient address to a Uint8Array of bytes.
+ *
```
Comment (wainola) [inline packages/sdk/src/utils/helpers.ts:122]: Could you add `@name` docs?

**Review by wainola — CHANGES_REQUESTED** [review]

**Review by MakMuftic — DISMISSED** [review]

File: packages/sdk/src/chains/EVM/fee/basicFee.ts [hunk 2]
```diff
@@ -41,7 +41,7 @@ export const calculateBasicfee = async ({
   tokenAmount: string;
   recipientAddress: string;
 }): Promise<FeeDataResult> => {
-  const depositData = constructDepositDataEvmSubstrate(tokenAmount, recipientAddress);
+  const depositData = createERCDepositData(tokenAmount, recipientAddress);
   // WHY 0X00 AND NOT 0X0?
```
Comment (nmlinaric) [inline packages/sdk/src/chains/EVM/fee/basicFee.ts:45]: What's this? What do you want to do exactly?

File: packages/sdk/src/utils/__test__/helpers.test.ts [hunk 3]
```diff
@@ -30,16 +38,50 @@ describe('constructDepositDataEvmSubstrate', () => {
...
-    expect(utils.isAddress).toHaveBeenCalledWith(recipientAddress);
-    expect(utils.arrayify).toHaveBeenCalledWith(recipientAddress);
+    expect(helpers.getRecipientAddressInBytes).toHaveBeenCalledWith(recipientAddress);
     expect(utils.hexlify).toHaveBeenCalledWith(expectedBytesArr);
     expect(depositData).toEqual(expectedDepositData);
   });
 });
+
+describe('getRecipientAddressInBytes', () => {
+  it('should convert an EVM address to a Uint8Array of bytes', () => {
```
Comment (nmlinaric) [inline packages/sdk/src/utils/__test__/helpers.test.ts:56]: `getRecipientAddressInBytes(evmAddress)` basically does `utils.arrayify(evmAddress)` so if the test is called convert to `Uint8Array` it should be something like:
```
result = getRecipientAddressInBytes(evmAddress)
expect(result).isinstanceof(Uint8Array).to.be.true;
```

File: packages/sdk/src/utils/__test__/helpers.test.ts [hunk 4]
```diff
@@ -30,16 +38,50 @@ describe('constructDepositDataEvmSubstrate', () => {
...
+    const evmAddress = '0x742d35Cc6634C0532925a3b844Bc454e4438f44e';
+    expect(utils.isAddress(evmAddress)).toBeTruthy();
+
+    const result = getRecipientAddressInBytes(evmAddress);
+    const expectedResult = utils.arrayify(evmAddress);
+
+    expect(result).toEqual(expectedResult);
+  });
+
+  it('should convert a Substrate multilocation to a Uint8Array of bytes', () => {
```
Comment (nmlinaric) [inline packages/sdk/src/utils/__test__/helpers.test.ts:66]: Same thing as above, and I assume `0x06a220edf5f82b84fc5f9270f8a30a17636bf29c05a5c16279405ca20918aa39` is substrate equivalent of evm `0x742d35Cc6634C0532925a3b844Bc454e4438f44e` and I don't see the logic that does that coversion.

**Review by nmlinaric — CHANGES_REQUESTED** [review]

File: packages/sdk/src/utils/__test__/helpers.test.ts (hunk 3 above)
Comment (enemycnt) [reply packages/sdk/src/utils/__test__/helpers.test.ts:56]: added `expect(result).toBeInstanceOf(Uint8Array);`

File: packages/sdk/src/utils/__test__/helpers.test.ts (hunk 4 above)
Comment (enemycnt) [reply packages/sdk/src/utils/__test__/helpers.test.ts:67]: I do not convert addresses here. I convert json-like object of multilocation to string, and then to bytes. Then these data will be hexified with other params and passed as depositData to the contract.

Comment (enemycnt) [reply packages/sdk/src/utils/__test__/helpers.test.ts:67]: added `expect(result).toBeInstanceOf(Uint8Array);`. Same as above

File: packages/sdk/src/chains/EVM/fee/basicFee.ts (hunk 2 above)
Comment (wainola) [reply packages/sdk/src/chains/EVM/fee/basicFee.ts:45]: no, we can get rid of it

**Review by wainola — APPROVED** [review]

**Review by wainola — DISMISSED** [review]

File: packages/sdk/src/utils/__test__/helpers.test.ts (hunk 4 above)
Comment (enemycnt) [reply packages/sdk/src/utils/__test__/helpers.test.ts:67]: It is public key of substrate address

Comment (enemycnt) [reply packages/sdk/src/utils/__test__/helpers.test.ts:67]: The address `5CDQJk6kxvBcjauhrogUc9B8vhbdXhRscp1tGEUmniryF1Vt` equals to `0x06a220edf5f82b84fc5f9270f8a30a17636bf29c05a5c16279405ca20918aa39`
https://stackoverflow.com/questions/60983883/how-to-encode-the-hex-string-representation-of-an-account-id-in-substrate-using

Comment (enemycnt) [reply packages/sdk/src/utils/__test__/helpers.test.ts:67]: @freddyli7 please correct me if use wrong terminology here

**Review by wainola — APPROVED** [review]

**Pull request merged**